### PR TITLE
Fix fee of CreateMultisigAccount

### LINKED ISSUE #2710: GAS price for CreateMultisigAccount is too cheap: 14 mins with 20GAS
The syscall `System.Contract.CreateMultisigAccount` is based on `Sha256` while its price is lower than `CryptoLib.Sha256`'s (`1<<10` < `1<<15`).

Therefore, a minor DOS can be composed using CreateMultisigAccount syscall.

```
curl http://seed1t5.neo.org:20332 -d '{ "jsonrpc": "2.0", "id": 1, "method": "invokescript", "params": ["VwIANSUAAAB3AAKIcSMAdwFvABFBajPpCUlvAZ13AW8BJfD///9JQFcCAAwhAswQ0OkpynUs/TQIvt2gZGPi2T/UNeTCuGqJWzeS3uTIAf8DdwFvASMNAAAASm8BnXcBbwEl+P///0PAdwBvAEA="] }'| json_pp
```

source code: https://github.com/lazynode/Tanya/pull/26/files

I propose that charge `CreateMultisigAccount` as `Neo.Crypto.CheckMultisig` does which is dynamic and depends on accnout number.

### PR DESCRIPTION
**Pull request by erikzhang**
Fix #2710

### CODE REVIEW

File: src/neo/SmartContract/ApplicationEngine.Contract.cs [hunk 1]
```diff
@@ -132,8 +132,9 @@ internal protected static UInt160 CreateStandardAccount(ECPoint pubKey)
         /// <param name="m">The minimum number of correct signatures that need to be provided in order for the verification to pass.</param>
         /// <param name="pubKeys">The public keys of the account.</param>
         /// <returns>The hash of the account.</returns>
-        internal protected static UInt160 CreateMultisigAccount(int m, ECPoint[] pubKeys)
+        internal protected UInt160 CreateMultisigAccount(int m, ECPoint[] pubKeys)
         {
+            AddGas(CheckSigPrice * pubKeys.Length * ExecFeeFactor);
```
Comment (roman-khimov) [inline src/neo/SmartContract/ApplicationEngine.Contract.cs:137]: This "more keys - more GAS" dependency looks correct anyway.

**Review by roman-khimov — COMMENTED** [review]
Anyone tested it already wrt mainnet compatibility?

**Review by shargon — APPROVED** [review]
Need to be checked if it was used in mainnet. But code looks good

**Comment by Liaojinghui** [comment]
I agree with @shargon, changes like this have to be cautious cause it might cause new node fail to synchronize.

**Comment by superboyiii** [comment]
I synced a mainnet node based on this, it stopped at 1353442.
I also compared the tx in that block, it's a flamingo swap transaction and the execution result differed a lot. Should be influenced by previous tx which contains `CreateMultisigAccount`. I think we need add an update height for this.
```
v3.2.1

{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "txid": "0x1717a1569a584cbd566d16f035dee66e6da495bffd73a11ca0c38c9447cc38cd",
        "executions": [
            {
                "trigger": "Application",
                "vmstate": "HALT",
                "exception": null,
                "gasconsumed": "26674620",
                "stack": [
                    {
                        "type": "Boolean",
                        "value": true
                    },
                    {
                        "type": "Array",
                        "value": [
                            {
                                "type": "Integer",
                                "value": "34979980"
                            },
                            {
                                "type": "Integer",
                                "value": "3571375614"
                            },
                            {
                                "type": "Integer",
                                "value": "350845763"
                            }
                        ]
                    },
                    {
                        "type": "Boolean",
                        "value": true
                    }
                ],
                "notifications": [
                    {
                        "contract": "0x48c40d4666f93408be1bef038b6722404d9a4c2a",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "7MAgTg6XM6llts9y33c3yLCFWk0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "35261524"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0xf0151f528127558851b39c2cd8aa47da7418ab28",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "7MAgTg6XM6llts9y33c3yLCFWk0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "3589322225"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Synced",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "Integer",
                                    "value": "63198548054442"
                                },
                                {
                                    "type": "Integer",
                                    "value": "6452426554022242"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Swapped",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "I0/XJYzDXXchuDI3tmXX7Mz0cPk="
                                },
                                {
                                    "type": "Integer",
                                    "value": "35261524"
                                },
                                {
                                    "type": "Integer",
                                    "value": "0"
                                },
                                {
                                    "type": "Integer",
                                    "value": "0"
                                },
                                {
                                    "type": "Integer",
                                    "value": "3589322225"
                                },
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x48c40d4666f93408be1bef038b6722404d9a4c2a",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "7MAgTg6XM6llts9y33c3yLCFWk0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "34979980"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0xf0151f528127558851b39c2cd8aa47da7418ab28",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "7MAgTg6XM6llts9y33c3yLCFWk0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "3571375614"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "Any"
                                },
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "350845763"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Synced",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "Integer",
                                    "value": "63198583034422"
                                },
                                {
                                    "type": "Integer",
                                    "value": "6452430125397856"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Minted",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "I0/XJYzDXXchuDI3tmXX7Mz0cPk="
                                },
                                {
                                    "type": "Integer",
                                    "value": "34979980"
                                },
                                {
                                    "type": "Integer",
                                    "value": "3571375614"
                                },
                                {
                                    "type": "Integer",
                                    "value": "350845763"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "Y2/EnjqWCjTyTe8vMvZAGY73qdE="
                                },
                                {
                                    "type": "Integer",
                                    "value": "349091337"
                                }
                            ]
                        }
                    }
                ]
            }
        ]
    }
}
```

```
this PR:

{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "txid": "0x1717a1569a584cbd566d16f035dee66e6da495bffd73a11ca0c38c9447cc38cd",
        "executions": [
            {
                "trigger": "Application",
                "vmstate": "HALT",
                "exception": null,
                "gasconsumed": "26674620",
                "stack": [
                    {
                        "type": "Boolean",
                        "value": true
                    },
                    {
                        "type": "Array",
                        "value": [
                            {
                                "type": "Integer",
                                "value": "34979980"
                            },
                            {
                                "type": "Integer",
                                "value": "3571375614"
                            },
                            {
                                "type": "Integer",
                                "value": "350845763"
                            }
                        ]
                    },
                    {
                        "type": "Boolean",
                        "value": true
                    }
                ],
                "notifications": [
                    {
                        "contract": "0x48c40d4666f93408be1bef038b6722404d9a4c2a",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "7MAgTg6XM6llts9y33c3yLCFWk0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "35261524"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0xf0151f528127558851b39c2cd8aa47da7418ab28",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "7MAgTg6XM6llts9y33c3yLCFWk0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "3589322225"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Synced",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "Integer",
                                    "value": "63198548054442"
                                },
                                {
                                    "type": "Integer",
                                    "value": "6452426554022242"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Swapped",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "I0/XJYzDXXchuDI3tmXX7Mz0cPk="
                                },
                                {
                                    "type": "Integer",
                                    "value": "35261524"
                                },
                                {
                                    "type": "Integer",
                                    "value": "0"
                                },
                                {
                                    "type": "Integer",
                                    "value": "0"
                                },
                                {
                                    "type": "Integer",
                                    "value": "3589322225"
                                },
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x48c40d4666f93408be1bef038b6722404d9a4c2a",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "7MAgTg6XM6llts9y33c3yLCFWk0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "34979980"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0xf0151f528127558851b39c2cd8aa47da7418ab28",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "7MAgTg6XM6llts9y33c3yLCFWk0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "3571375614"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "Any"
                                },
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "Integer",
                                    "value": "350845763"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Synced",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "Integer",
                                    "value": "63198583034422"
                                },
                                {
                                    "type": "Integer",
                                    "value": "6452430125397856"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Minted",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "I0/XJYzDXXchuDI3tmXX7Mz0cPk="
                                },
                                {
                                    "type": "Integer",
                                    "value": "34979980"
                                },
                                {
                                    "type": "Integer",
                                    "value": "3571375614"
                                },
                                {
                                    "type": "Integer",
                                    "value": "350845763"
                                }
                            ]
                        }
                    },
                    {
                        "contract": "0x4d5a85b0c83777df72cfb665a933970e4e20c0ec",
                        "eventname": "Transfer",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "ByteString",
                                    "value": "kfUEM8/TPomnhlDaMiNbj9rlbz0="
                                },
                                {
                                    "type": "ByteString",
                                    "value": "Y2/EnjqWCjTyTe8vMvZAGY73qdE="
                                },
                                {
                                    "type": "Integer",
                                    "value": "349091337"
                                }
                            ]
                        }
                    }
                ]
            }
        ]
    }
}
```
@shargon @erikzhang @roman-khimov

**Comment by Liaojinghui** [comment]
We should name every hard fork, to make it more clear which fork we are working on. 

Not much change to make based on this pr, just add an enum with a list of hardfork names (including names of future harforks) then match them with the hardfork versions would work.

**Comment by superboyiii** [comment]
> We should name every hard fork, to make it more clear which fork we are working on.
> 
> Not much change to make based on this pr, just add an enum with a list of hardfork names (including names of future harforks) then match them with the hardfork versions would work.

Good idea.

**Comment by roman-khimov** [comment]
> We should name every hard fork, to make it more clear which fork we are working on.

There is also a use case for private networks with "whatever is the latest" hard fork enabled, it should be easy to configure that too.

**Comment by erikzhang** [comment]
> We should name every hard fork, to make it more clear which fork we are working on.

Done.

> There is also a use case for private networks with "whatever is the latest" hard fork enabled, it should be easy to configure that too.

Agree. Now you can just delete `Hardfork` section or leave its contents empty to configure a "whatever is the latest" private network.

**Comment by roman-khimov** [comment]
BTW, if we're doing this then maybe we can make T4 testnet compatible with 3.2.2? It only requires making #2653 change fork-dependent (see #2673).

**Review by shargon — APPROVED** [review]
Code looks good to me, good change!

**Comment by erikzhang** [comment]
@shargon Can you revert the commit "[T4 testnet compatible with 3.2.2](https://github.com/neo-project/neo/pull/2712/commits/15bea615e36a368ef6f27064aa16f247feaeef75)"? We can discuss it in a new issue or pr.

**Review by shargon — APPROVED** [review]

**Pull request merged**